clippy_lints/src/utils/hir_utils.rs

```diff
@@ -667,10 +706,15 @@ impl<'a, 'tcx> SpanlessHash<'a, 'tcx> {
         // self.maybe_typeck_results.unwrap().qpath_res(p, id).hash(&mut self.s);
     }
 
-    pub fn hash_path(&mut self, p: &Path<'_>) {
-        p.is_global().hash(&mut self.s);
-        for p in p.segments {
-            self.hash_name(p.ident.name);
+    pub fn hash_path(&mut self, path: &Path<'_>) {
+        match path.res {
+            // constant hash since equality is dependant on inter-expression context
+            Res::Local(_) => 1_usize.hash(&mut self.s),
```

Comment:
This is not consistent with the `PartialEq` implementation. `Hash` assumes that all locals are the same, in the `PartialEq` impl they can be different.

Reply:
I believe this is ok since a hash function is allowed to have collisions.

Reply:
Just read the [`Hash` documentation](https://doc.rust-lang.org/std/hash/trait.Hash.html#hash-and-eq) and only `Eq` -> `Hash` has to hold up, not `Hash` -> `Eq`. For some reason I thought, that `Hash <-> Eq`. In that case this should be fine.